flow/flow.go

```diff
@@ -319,6 +319,35 @@ func addSegment(in *low.Ring, first *Func) *processSegment {
 	return segment
 }
 
+type HWCapability int
+
+const (
+	HWTXChecksumCapability HWCapability = iota
+)
+
+// CheckHWCapability return true if hardware offloading capability
+// present in all ports. Otherwise it returns false.
+func CheckHWCapability(capa HWCapability, ports []uint16) bool {
+	for p := range ports {
+		switch capa {
+		case HWTXChecksumCapability:
+			if !low.CheckHWTXChecksumCapability(ports[p]) {
```

Comment:
As I understand capa can have only one value, if it is right - you can just return low level function output without any checking.

Reply:
Other values are supposed to be added in the future, so it makes sense to write the handling code from the beginning.